spec/compiler/semantic/tuple_spec.cr

```diff
@@ -287,7 +287,18 @@ describe "Semantic: tuples" do
         pos += {0, 0}
       end
       ),
-      "tuple too big"
+      "tuple size cannot be greater than 300 (size is 301)"
+  end
+
+  it "errors on named tuple too big" do
+    assert_error %(
+      { #{String.build do |io|
```

Comment:
This interpolation is not needed. And you could assign the `String.build` to a variable `code` and use it. This way is too convoluted.

Reply:
Oh, it seems it's actually needed because of the `{`. However, I still recommend building the code somewhere else. It'll be easier to understand.

Reply:
The interpolation could just be `#{ 333.times.map{|i| "arg#{i}: 0,"}.join }`.